Tests/YamsTests/EncoderTests.swift

```diff
@@ -0,0 +1,1006 @@
+//
+//  EncoderTests.swift
+//  Yams
+//
+//  Created by Norio Nomura on 5/2/17.
+//  Copyright (c) 2017 Yams. All rights reserved.
+//
+
+import Foundation
+import XCTest
+import Yams
+
+#if swift(>=4.0)
+
+    // swiftlint:disable identifier_name line_length
+
+    /// Tests are copied from https://github.com/apple/swift/blob/master/test/stdlib/TestJSONEncoder.swift
+    class EncoderTests: XCTestCase {
+        // MARK: - Encoding Top-Level Empty Types
+        func testEncodingTopLevelEmptyStruct() {
+            let empty = EmptyStruct()
+            _testRoundTrip(of: empty, expectedYAML: "{}\n")
+        }
+
+        func testEncodingTopLevelEmptyClass() {
+            let empty = EmptyClass()
+            _testRoundTrip(of: empty, expectedYAML: "{}\n")
+        }
+
+        // MARK: - Encoding Top-Level Single-Value Types
+        func testEncodingTopLevelSingleValueEnum() {
+            _testRoundTrip(of: Switch.off, expectedYAML: "false\n...\n")
+            _testRoundTrip(of: Switch.on, expectedYAML: "true\n...\n")
+        }
+
+        func testEncodingTopLevelSingleValueStruct() {
+            _testRoundTrip(of: Timestamp(3141592653), expectedYAML: "3.141592653e+9\n...\n")
+        }
+
+        func testEncodingTopLevelSingleValueClass() {
+            _testRoundTrip(of: Counter(), expectedYAML: "0\n...\n")
+        }
+
+        // MARK: - Encoding Top-Level Structured Types
+        func testEncodingTopLevelStructuredStruct() {
+            // Address is a struct type with multiple fields.
+            let address = Address.testValue
+            _testRoundTrip(of: address, expectedYAML: """
+                street: 1 Infinite Loop
+                city: Cupertino
+                state: CA
+                zipCode: 95014
+                country: United States
+
+                """)
+        }
+
+        func testEncodingTopLevelStructuredClass() {
+            // Person is a class with multiple fields.
+            let person = Person.testValue
+            _testRoundTrip(of: person, expectedYAML: "name: Johnny Appleseed\nemail: appleseed@apple.com\n")
+        }
+
+        func testEncodingTopLevelStructuredSingleStruct() {
+            // Numbers is a struct which encodes as an array through a single value container.
+            let numbers = Numbers.testValue
+            _testRoundTrip(of: numbers, expectedYAML: "- 4\n- 8\n- 15\n- 16\n- 23\n- 42\n")
+        }
+
+        func testEncodingTopLevelStructuredSingleClass() {
```

Comment:
This test is flaky on Linux in my testing, where the order of keys doesn't appear to be stable across invocations.

```
EncoderTests.testEncodingTopLevelStructuredSingleClass : XCTAssertEqual failed: ("localhost:
  relative: http://127.0.0.1
Apple:
  relative: http://apple.com
") is not equal to ("Apple:
  relative: http://apple.com
localhost:
  relative: http://127.0.0.1
") - Produced YAML not identical to expected YAML.
```

Reply:
I'll add `sortedKeys` option to `Emitter.Options`.

Reply:
This will be PR against master.